sdk/monitor/monitor-opentelemetry/test/internal/unit/shared/config.test.ts

```diff
@@ -441,40 +275,39 @@ describe("Library/Config", () => {
     });
 
     it("instrumentation key validation-valid key passed", () => {
-      const warnStub = sandbox.stub(console, "warn");
-      const config = new InternalConfig();
-      config.azureMonitorExporterOptions.connectionString =
-        "InstrumentationKey=1aa11111-bbbb-1ccc-8ddd-eeeeffff3333";
-      assert.ok(warnStub.notCalled, "warning was not raised");
+      new InternalConfig({
+        azureMonitorExporterOptions: {
+          connectionString: "InstrumentationKey=1aa11111-bbbb-1ccc-8ddd-eeeeffff3333",
+        },
+      });
+      expect(warnStub).not.toHaveBeenCalled();
     });
 
-    it("instrumentation key validation-invalid key passed", () => {
-      const warnStub = sandbox.stub(console, "warn");
-      const config = new InternalConfig();
-      config.azureMonitorExporterOptions.connectionString =
-        "InstrumentationKey=1aa11111bbbb1ccc8dddeeeeffff3333";
-      assert.ok(warnStub.calledOn, "warning was raised");
+    // TODO: these tests are incorrect on main
```

Comment:
Just calling attention to this - the tests on `main` are incorrect and have been pended as part of this PR. I would expect them to be fixed by the package owners who are more familiar with the inner-workings.

Basically:

```ts
assert.ok(warnStub.calledOn); // ok
assert.ok(warnStub.calledOnce); // fails on main
```

This is because `calledOn` is a function, not a property, and so as a property it is _always_ truthy.

To validate: on main change the expection to `assert.ok(warnStub.calledOnce)` and watch it fail